test/es6/destructuring_catch.js

```diff
@@ -189,6 +189,164 @@ var tests = [
             }
         })();
      }
+   },
+   {
+        name: "Function definitions in catch's parameter",
+        body: function () {
+            (function() {
+                try {
+                    var c = 10;
+                    throw ['inside'];
+                } catch ([x, y = function() { return c; }]) {
+                    assert.areEqual(y(), 10, "Function should be able to capture symbols from try's body properly");
+                    assert.areEqual(x, 'inside', "Function should be able to capture symbols from try's body properly");
+                } 
+            })();
+
+            (function() {
+                try {
+                    throw [];
+                } catch ([x = 10, y = function() { return x; }]) {
+                    assert.areEqual(y(), 10, "Function should be able to capture symbols from catch's param");
+                } 
+            })();
+
+            (function() {
+                try {
+                    throw [];
+                } catch ([x = 10, y = function() { return x; }]) {
+                    eval("");
+                    assert.areEqual(y(), 10, "Function should be able to capture symbols from catch's param");
+                } 
+            })();
+
+            (function() {
+                try {
+                    throw {};
+                } catch ({x = 10, y = function() { return x; }}) {

```

Comment:
what happens if you have eval in the function, which is defined at the catch param ?